embedded-graphics/src/primitives/mod.rs

```diff
@@ -95,36 +114,57 @@ macro_rules! egcircle {
 /// Note that only the `stroke` property has any effect on lines currently.
 ///
 /// ```rust
-/// use embedded_graphics::prelude::*;
-/// use embedded_graphics::{egline, pixelcolor::Rgb565, primitives::Line};
-/// use embedded_graphics::style::{PrimitiveStyle, Styled};
-///
-/// let line: Styled<Line, PrimitiveStyle<Rgb565>> = egline!((10, 20), (30, 40));
-/// let line: Styled<Line, PrimitiveStyle<Rgb565>> = egline!(Point::new(10, 20), Point::new(30, 40));
-/// let stroke_line: Styled<Line, PrimitiveStyle<Rgb565>> = egline!((10, 20), (30, 40), stroke_color = Some(Rgb565::BLUE));
+/// use embedded_graphics::{
+///     egline,
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitive_style,
+///     primitives::Line,
+///     style::{PrimitiveStyle, Styled},
+/// };
+///
+/// let line: Styled<Line, PrimitiveStyle<Rgb565>> = egline!(start = (10, 20), end = (30, 40));
+/// let line: Styled<Line, PrimitiveStyle<Rgb565>> =
+///     egline!(start = Point::new(10, 20), end = Point::new(30, 40));
+/// let stroke_line: Styled<Line, PrimitiveStyle<Rgb565>> = egline!(
+///     start = (10, 20),
+///     end = (30, 40),
+///     style = primitive_style!(stroke_color = Rgb565::BLUE)
+/// );
 /// ```
 ///
-/// Style properties like `stroke_color` map to the properties in the [`PrimitiveStyle`] struct.
+/// Style properties like `stroke_color` map to methods on the [`PrimitiveStyleBuilder`] struct.
 /// For example, the following code makes two identical lines:
 ///
-/// [`PrimitiveStyle`]: style/struct.PrimitiveStyle.html
+/// [`PrimitiveStyleBuilder`]: style/struct.PrimitiveStyleBuilder.html
 ///
 /// ```rust
-/// use embedded_graphics::prelude::*;
-/// use embedded_graphics::{egline, pixelcolor::Rgb565, primitives::Line};
-/// use embedded_graphics::style::{PrimitiveStyle, PrimitiveStyleBuilder, Styled};
-///
-/// let line: Styled<Line, PrimitiveStyle<Rgb565>> = egline!(
-///     Point::new(10, 20),
-///     Point::new(30, 40),
-///     stroke_color = Some(Rgb565::BLUE),
-///     fill_color = Some(Rgb565::YELLOW)
+/// use embedded_graphics::{
+///     egline,
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitive_style,
+///     primitives::Line,
+///     style::{PrimitiveStyle, PrimitiveStyleBuilder, Styled},
+/// };
+///
+/// let line_1: Styled<Line, PrimitiveStyle<Rgb565>> = egline!(
+///     start = Point::new(10, 20),
+///     end = Point::new(30, 40),
+///     style = primitive_style!(
+///         stroke_color = Rgb565::BLUE,
+///         fill_color = Rgb565::YELLOW,
+///         stroke_width = 1
+///     )
```

Comment:
This should be removed. IMO the ability to use different point variants should be explained once in a prominent location and not clutter every example that tries to illustrate something else.

Reply:
I could add an explanation to the top of each macro's doc comment? A bit repetitive but I can't think of anywhere better.

Reply:
Adding some module level docs to the `primitives` module could provide an overview of how primitives can be created and drawn. And this could also mention the macros in general and how points can be defined when calling them.